_posts/2017-07-29-ruby-oop.md

```diff
@@ -0,0 +1,383 @@
+---
+layout: post
+author: z64
+title: Beginning OOP with Ruby
+date: 2017-07-29
+tags: ruby oop tutorial
+---
+
+This post attempts to guide a beginner through object-oriented progamming in Ruby, showing them the thought process behind both designing and interacting with a library, and problem solving in general with Ruby.
+
+To accomplish this, we will be solving a simple geometry problem:
+
+- Describe a collection of objects that can calculate the perimeter of a polygon with an arbitrary number of sides.
+
+## Breaking it down
+
+The first thing we want to do to tackle this problem is to break apart the problem into a series of smaller, bite sized questions. "Good Ruby code" will read a lot like English - I think you'll find the syntax we end up writing will almost look like English answers to these questions. You probably already know the answers to these (which is why I picked this as an example), but read them and pay attention to the language that I use.
+
+**1) What is perimeter?**
+
+> Perimeter is the total distance around an object, in our case a 2D polygon with any number of sides.
+
+**2) What is a Polygon?**
```

Comment:
for all 4 of these points, we should keep consistent about capitalising the subjects. Either lowercase as speaking about them in general cases, or uppercase as preparation to refer to them as a class object. Also, we can use inline codeblocks if that is the intention.